### PR TITLE
fix(contracts/test): fix goerli fork tests

### PR DESCRIPTION
**Pull request by kevinhalliday**
Fix goerli fork tests.

- fix goerli fork "deployment"
- allow tests on goerli strategies (like stETH and rETH)
- handle strategy shares vs. underlying
- add avs fork tests make target

task: none

### CODE REVIEW

File: contracts/foundry.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ fs_permissions = [
 
 [fuzz]
 seed = "0x6eed"
+runs = 10
```
Comment (ttarsi) [inline contracts/foundry.toml:22]: just 10?

File: contracts/script/avs/DeployGoerliAVS.s.sol [hunk 2]
```diff
@@ -52,13 +50,25 @@ contract DeployGoerliAVS is Script {
...
                 allowlist,
                 StrategyParams.goerli()
             )
         );
 
         return proxy;
     }
+
+    /// @dev deploy OmniAVS, but with a prank. necessary because we cannot
+    //       vm.startPrank() outside of this contract
```
Comment (ttarsi) [inline contracts/script/avs/DeployGoerliAVS.s.sol:64]: I thought if `vm.startPrank` is called, all txns are called with that address until stopped – regardless of where the calls happen

**Review by ttarsi — APPROVED** [review]

File: contracts/script/avs/DeployGoerliAVS.s.sol (hunk 2 above)
Comment (kevinhalliday) [reply contracts/script/avs/DeployGoerliAVS.s.sol:64]: I did too so I wrote it that way first. But it didn't work. It's just all calls from the contract in which `vm.startPrank` was called. But that actually makes sense. You don't want `vm.startPrank()` to affect all `msg.sender`s in a chain of function calls

So

```
contract A {
     function foo() {
        vm.startPrank(me);
        b.foo();
    }
}

contract B {
    function bar() {
        // msg.sender == me
        c.baz()
   }
}


contract C {
   function baz() {
       // msg.sender == address(b)
   }
}
```

File: contracts/foundry.toml (hunk 1 above)
Comment (kevinhalliday) [reply contracts/foundry.toml:22]: Did this cause fork tests were very slow let me change just to do 10 runs for fork tests

**Pull request merged**